src/lib/environment.js

```diff
@@ -25,5 +30,16 @@ module.exports = {
   get instanceUrl() { return this.apiUrl && this.apiUrl.replace(/\/medic$/, ''); },
   get extraArgs() { return state.extraArgs; },
   get apiUrl() { return state.apiUrl; },
-  get force() { return state.force; }
+  get force() { return state.force; },
+
+  /**
+   * Return `true` if the environment seems to be production.
+   * @returns {boolean}
+   */
+  isProduction() {
+    if (!this.instanceUrl) return false;
+    const hostname = url.parse(this.instanceUrl).hostname;
+    if (LOCAL_MATCHER.test(hostname)) return false;
```

Comment:
Nitpick: for consistency use braces and new lines (this line, and line 40), eg:

```suggestion
    if (LOCAL_MATCHER.test(hostname)) {
      return false;
    }
```

I know this file has examples of the one line style, but it's inconsistent with the rest of the codebase.

Another option would be something like...

```suggestion
    return !LOCAL_MATCHER.test(hostname) &&
      !hostname.split('.').some(subdomain => DEV_MATCHER.test(subdomain));
```